crates/trie/sparse/src/either.rs

```diff
@@ -0,0 +1,141 @@
+//! An either implementation for the [`SparseTrieInterface`]
+
+use core::fmt::Debug;
+
+use crate::{
+    blinded::BlindedProvider, LeafLookup, LeafLookupError, SparseTrieInterface, SparseTrieUpdates,
+    TrieMasks,
+};
+use alloc::vec::Vec;
+use alloy_primitives::B256;
+use reth_execution_errors::SparseTrieResult;
+use reth_trie_common::{Nibbles, TrieNode};
+
+/// An `Either` type for types that implement the sparse trie interface.
+#[derive(Debug)]
+pub enum Either<A, B> {
+    /// The left sparse trie
+    Left(A),
+    /// The right sparse trie
+    Right(B),
+}
```

Comment:
should we impl this directly for either::Either ?

Reply:
hmm yeah let's just do it for either::Either